### PR TITLE
MIVisionX compatibility mods for cpu/hip/ocl

### PR DESCRIPTION
**Pull request by r-abishek**
Makes necessary changes in cpu/hip/ocl builds for MIVisionX compatibility.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -84,6 +84,11 @@ if( "${BACKEND}" STREQUAL "HIP" OR "${BACKEND}" STREQUAL "hip")
     find_package(HIP QUIET)    # Find package HIP
     if(HIP_FOUND)
 
+        # Set external flag to notify GPU_SUPPORT and HIP backend availability - Compatible with rocAL and MIVisionX
+        add_definitions(-DGPU_SUPPORT)
+        set(RPP_BACKEND_HIP 1) # For cmakedefine01 in rpp_backend.h.in
```
Comment (rrawther) [inline CMakeLists.txt:89]: don't you need to set RPP_BACKEND_OPENCL 0 here

File: CMakeLists.txt [hunk 2]
```diff
@@ -109,6 +114,11 @@ elseif("${BACKEND}" STREQUAL "OCL" OR "${BACKEND}" STREQUAL "OPENCL" OR "${BACKE
     find_package(OpenCL QUIET)    # Find package OpenCL
     if(OpenCL_FOUND)
 
+        # Set external flag to notify GPU_SUPPORT and OPENCL backend availability - Compatible with rocAL and MIVisionX
+        add_definitions(-DGPU_SUPPORT)
+        set(RPP_BACKEND_OPENCL 1) # For cmakedefine01 in rpp_backend.h.in
```
Comment (rrawther) [inline CMakeLists.txt:119]: don't you need to set RPP_BACKEND_HIP 0 here

File: include/rpp_backend.h.in [hunk 3]
```diff
@@ -1,8 +1,7 @@
 #ifndef GUARD_RPP_BACKEND_H_IN
 #define GUARD_RPP_BACKEND_H_IN
 
-#cmakedefine01 GPU_SUPPORT
```
Comment (rrawther) [inline include/rpp_backend.h.in:4]: why is the GPU_SUPPORT removed from here?

File: utilities/rpp-performancetests/HIP_NEW/CMakeLists.txt [hunk 4]
```diff
@@ -28,7 +28,7 @@ if (hip_FOUND)
     add_executable(Tensor_hip_pln3 Tensor_hip_pln3.cpp)
     # add_executable(Single_hip Single_hip.cpp)
     add_executable(uniqueFunctionalities_hip uniqueFunctionalities_hip.cpp)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp -DGPU_SUPPORT=1 -DHIP_COMPILE=1 -std=c++11")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp -DGPU_SUPPORT=1 -DRPP_BACKEND_HIP=1 -std=c++11")
     target_link_libraries(BatchPD_hip_pkd3 ${OpenCV_LIBS} -lamd_rpp ${HIP_LIBRARIES} pthread boost_filesystem boost_system hip::host)
```
Comment (rrawther) [inline utilities/rpp-performancetests/HIP_NEW/CMakeLists.txt:32]: link libraries for the hip is hip::host which might make it run on host (please double check). I think it should be hip::device

**Review by rrawther — COMMENTED** [review]
please address the review comments

File: CMakeLists.txt (hunk 1 above)
Comment (r-abishek) [reply CMakeLists.txt:89]: I tested it out and #cmakedefine01 seems to define RPP_BACKEND_OPENCL automatically to 0 and the installed include/rpp_backend.h file under /opt/rocm/rpp/include looks like this for the HIP install:
```
#ifndef GUARD_RPP_BACKEND_H_IN
#define GUARD_RPP_BACKEND_H_IN

#define RPP_BACKEND_HIP 1
#define RPP_BACKEND_OPENCL 0

#endif
```

File: CMakeLists.txt (hunk 2 above)
Comment (r-abishek) [reply CMakeLists.txt:119]: Similarly, the include/rpp_backend.h looks like this for the OpenCL install:
```
#ifndef GUARD_RPP_BACKEND_H_IN
#define GUARD_RPP_BACKEND_H_IN

#define RPP_BACKEND_HIP 0
#define RPP_BACKEND_OPENCL 1

#endif
```

File: include/rpp_backend.h.in (hunk 3 above)
Comment (r-abishek) [reply include/rpp_backend.h.in:4]: The builds seemed to work without it

**Comment by kiritigowda** [comment]
@r-abishek is this ready to be merged?

**Comment by r-abishek** [comment]
@kiritigowda Yeah this could be merged. This ensures build compatibility with MIVisionX

**Pull request merged**